openflow-sdn/openflow-protocol.md

```diff
@@ -0,0 +1,65 @@
+
+After this, it is said that the OpenFlow channel is successfully established between the switch and the controller. 
+
+The connection between the controller and switch is really important as it is the only way for a switch to communicate with a controller. To secure this connection, a protocol like TLS can also be used instead of a TCP connection. Here, the controller and switch need to have the proper certificates and keys for a successful TLS connection. This prevents snooping on the OF channel.
+
+### OpenFlow tables and Flow entries
+
+Flow tables are like a traditional switch's MAC/CAM table that stores the hardware addresses of hosts. Flow tables store flow entries or flows, that tell the SDN switch what to do with a packet when it comes on an incoming port.
+The switch will generally match certain parameters like IP address, port number, MAC address, VLAN ID, etc. and select the best matching flow entry from the table and execute the action associated with that entry. Actions could be to drop the packet, forward it out a different port, flood the packet or send it to the controller for further inspection of the packet.
+
+If a switch does not have an entry for a packet, the switch might have a default entry or "TABLE_MISS" entry. This entry has the lowest priority and the actions can either be to drop the packet or send it to the controller. When the controller receives this kind of packet from a switch, it sends it to the application running at the application layer which can process the packet and let the controller know if a new flow entry needs to be inserted in the switch's flow table. If that's the case, the controller will insert a flow entry on the switch. 
+
+The next packet of the same kind will be dealt with by the switch at the Data layer as it already has an entry and appropriate actions would be taken. This improves the efficiency of the network by a huge factor. 
+
+## Advantages of Using OpenFlow
+There are several advantages of OpenFlow and SDN rather than traditional networks.
+1. SDN enables separation of control and data plane which means switches can use all their hardware resources in just forwarding data instead of computing routes. 
+2. OpenFlow provides an easy way of communication between controller and switch which can be easily implemented in an existing network. 
+3. Most current devices in the field support OpenFlow, it is not enabled by default, but we can easily enable and use it in the transition to SDN. 
+4. It provides security with a TLS connection to prevent snooping and DoS attacks on the controller and/or the network. 
+5. OpenFlow does **NOT** change the configuration for a switch. It just updates the flow tables which define the path for a packet.
+
```

Comment:
```suggestion
There are several advantages of OpenFlow and SDN rather than traditional networks:

1. SDN enables separation of control and data plane, which means switches can use all their hardware resources in just forwarding data instead of computing routes. 
2. OpenFlow provides an easy way of communication between controller and switch, easily implemented in an existing network. 
3. Most current devices support OpenFlow, it is not enabled by default, but we can easily enable and use it in the transition to SDN. 
4. It provides security with a TLS connection to prevent snooping and DoS attacks on the controller and/or the network. 
5. OpenFlow does **NOT** change the configuration for a switch. It just updates the flow tables, which define the path for a packet.
```